src/test/java/itest/ShouldAcceptMultipartWithBoundaryIT.java

```diff
@@ -0,0 +1,172 @@
+                new JsonObject(
+                        Map.of(
+                                "meta",
+                                Map.of("type", HttpMimeType.JSON.mime(), "status", "OK"),
+                                "data",
+                                NULL_RESULT));
+
+        try {
+            JsonObject deleteResult = deleteResponse.get(5, TimeUnit.SECONDS);
+            MatcherAssert.assertThat(deleteResult, Matchers.equalTo(expectedDeleteResponse));
+        } catch (TimeoutException e) {
+            System.out.println("Deletion timed out. Reason: " + e.getMessage());
+            throw new RuntimeException("Deletion did not complete within the expected time.", e);
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof HttpException) {
+                HttpException httpException = (HttpException) e.getCause();
+                MatcherAssert.assertThat(httpException.getStatusCode(), Matchers.equalTo(400));
+                MatcherAssert.assertThat(
+                        httpException.getMessage(), Matchers.equalTo("Bad Request"));
+            } else {
+                System.out.println(
+                        "Deletion execution failed. Reason: " + e.getCause().getMessage());
+                throw e;
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            System.out.println("Deletion operation interrupted. Reason: " + e.getMessage());
+            throw e;
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "multipart/form-data",
+                "multipart/form-data; boundary=------somecharacters",
+                "multipart/form-data; unkown characters",
+                "multipart/form-data; directive1; directive2",
+                "multipart/form-data; directive"
+            })
+    void shouldAcceptMultipartWithBoundary(String contentType) throws Exception {
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.set("name", TEST_RULE_NAME);
+        form.set("matchExpression", "false");
+        form.set("eventSpecifier", "template=Continuous");
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        webClient
+                .post("/api/v2/rules")
+                .putHeader(HttpHeaders.CONTENT_TYPE.toString(), contentType)
+                .sendForm(
+                        form,
+                        ar -> {
+                            if (assertRequestStatus(ar, response)) {
+                                response.complete(ar.result().bodyAsJsonObject());
+                            } else {
+                                response.completeExceptionally(
+                                        new RuntimeException("Request failed"));
+                            }
+                        });
+        try {
+            JsonObject result = response.get(5, TimeUnit.SECONDS);
```

Comment:
I think there should be some assertions made about the response here, no?

Reply:
yes